.github/workflows/_build-wheel-release-upload.yml

```diff
@@ -22,47 +22,35 @@ on:
         required: true
 
 jobs:
-  tag-check:
-    runs-on: ubuntu-latest
-    outputs:
-      tag_valid: ${{ steps.tag-check.outputs.tag_valid }}
-    steps:
-      - name: Extract and validate tag for (pre)-release
-        id: tag-check
-        run: |
-          TAG_NAME=${GITHUB_REF#refs/tags/}
+  tag-privilege-check:
+    uses: ./.github/workflows/_release_tag_privilege_check.yml
+    
+  build-pure-python-package:
+    needs: [tag-privilege-check]
+    if: inputs.c_extension == false
+    uses: ./.github/workflows/_build-pure-python-package.yml
+    
+  build-non-pure-python-package:
+    needs: [tag-privilege-check]
+    if: inputs.project == 'diffpy.pdffit2'
```

Comment:
After tag check and also checking the name `diffpy.pdffit2`, build sdist and wheel files - again modularize the python build . The design here is scalable. We have to provide a build recipe for a package containing C++ code.